### PR TITLE
[ARCTIC-1185][Spark][Improvement]ArcticSparkSessionCatalog need fail back to hive when AMS is not available

### LINKED ISSUE #1185: [Spark][Improvement]: ArcticSparkSessionCatalog need fail back to hive when AMS is not available
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

When AMS is down, the spark engine can't load ArcticSparkSessionCatalog, and this may cause the other hive tables are not usable.

### How should we improve?

If the spark uses ArcticSparkSessionCatalog,  only Mixed-format tables are not available,  the native hive tables will not be influenced when AMS is not available.

This improvement needs to pick to 0.3.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?

resolve #1185 

When AMS is down, the spark engine can't load ArcticSparkSessionCatalog, and this may cause the other hive tables are not usable.

## Brief change log
1.  Remove the spark.sql.arctic.delegate.enabled could be removed.
2. All actions use session catalog to load table metadata. and only using arcticCatalog if we have been checked the table format type and have been made sure this is an ArcticTable

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
 import java.util.Map;
 
 import static com.netease.arctic.spark.SparkSQLProperties.DELEGATE_DEFAULT_CATALOG_TABLE;
+import static com.netease.arctic.spark.SparkSQLProperties.DELEGATE_DEFAULT_CATALOG_TABLE_DEFAULT;
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java:44]: I think the `delegate default catalog enabled` could be removed

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java [hunk 2]
```diff
@@ -116,14 +117,8 @@ public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceExcepti
...
-        return getArcticCatalog().loadTable(ident);
-      } catch (NoSuchTableException e) {
-        LOG.info("no arctic table, load hive table by default spark session catalog.");
-        return getSessionCatalog().loadTable(ident);
-      }
+    if (isDelegateEnable() && arcticTableExists(ident)) {
+      return getArcticCatalog().loadTable(ident);
     } else {
       return getSessionCatalog().loadTable(ident);
     }
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java:124]: code should like this

```
Table table = getSessionCatalog().loadTable(ident);
if( isArcticTable(table) ) {
   return arcticCatalog.loadTable( ident)
}
return table;
```

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java [hunk 3]
```diff
@@ -156,9 +151,13 @@ public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchT
   public boolean dropTable(Identifier ident) {
     // no need to check table existence to determine which catalog to use. if a table doesn't exist then both are
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java:152]:
```
Table table = loadTable(ident);
if( isArcticTable ) {
   arcticCatalog.alterTable(ident)
} else {
   sessionCatalog.alterTable( ident )
}
```

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java [hunk 4]
```diff
@@ -167,7 +166,7 @@ public boolean dropTable(Identifier ident) {
   public void renameTable(Identifier from, Identifier to) throws NoSuchTableException, TableAlreadyExistsException {
     // rename is not supported by HadoopCatalog. to avoid UnsupportedOperationException for session catalog tables,
     // check table existence first to ensure that the table belongs to the Iceberg catalog.
-    if (isDelegateEnable() && getArcticCatalog().tableExists(from)) {
+    if (isDelegateEnable() && arcticTableExists(from)) {
       getArcticCatalog().renameTable(from, to);
     } else {
       getSessionCatalog().renameTable(from, to);
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkSessionCatalog.java:172]: same like alter

**Review by baiyangtx — CHANGES_REQUESTED** [review]
1. I think remove the `spark.sql.arctic.delegate.enabled` could be removed.
2. all actions use session catalog to load table metadata. and only using arcticCatalog if we have been checked the table format type and have been made sure this is an ArcticTable

**Comment by zhoujinsong** [comment]
@hellojinsilei Could you please complete the PR's description as the Arctic Community Guidld：
https://docs.google.com/document/d/1Pwv1LflW3z5yq34qDpEYwSxOB3LIE5qoKQ_5Sv0IV1Y/edit?usp=sharing

**Review by baiyangtx — APPROVED** [review]

**Pull request merged**